### PR TITLE
Update AnalysisNexxim.py

### PR DESCRIPTION
**Pull request by PipKat**

### CODE REVIEW

File: pyaedt/application/AnalysisNexxim.py [hunk 1]
```diff
@@ -70,49 +68,51 @@ def __init__(self, application, projectname, designname, solution_type, setup_na
...
 
         Parameters
         ----------
-        excitation_names :
-            list of excitation to include (Default value = [])
-
+        eexcitation_names : list, optional
+            List of excitations. The default is ``[]``, in which case
+            the S parameters for all excitations are to be provided. 
+
```
Comment (PipKat) [inline pyaedt/application/AnalysisNexxim.py:106]:
```suggestion
            For example, ``["1","2"]``.
            
```

File: pyaedt/application/AnalysisNexxim.py [hunk 2]
```diff
@@ -70,49 +68,51 @@ def __init__(self, application, projectname, designname, solution_type, setup_na
...
+        eexcitation_names : list, optional
+            List of excitations. The default is ``[]``, in which case
+            the S parameters for all excitations are to be provided. 
+            
         Returns
         -------
-        type
-            list of strin representing Sparameters of excitations
+        list
+            List of strings representing the S parameters of the excitations.
```
Comment (PipKat) [inline pyaedt/application/AnalysisNexxim.py:110]:
```suggestion
            List of strings representing the S parameters of the excitations.
            For example, ``"S(1,1)", "S(1,2)", S(2,2)``.
```

File: pyaedt/application/AnalysisNexxim.py [hunk 3]
```diff
@@ -70,49 +68,51 @@ def __init__(self, application, projectname, designname, solution_type, setup_na
...
+            
+        Examples
```
Comment (PipKat) [inline pyaedt/application/AnalysisNexxim.py:112]:
```suggestion
```

File: pyaedt/application/AnalysisNexxim.py [hunk 4]
```diff
@@ -70,49 +68,51 @@ def __init__(self, application, projectname, designname, solution_type, setup_na
...
+        --------
+        
+        >>>excitation_names ["1","2"] output ["S(1,1)", "S(1,2)", S(2,2)]
```
Comment (PipKat) [inline pyaedt/application/AnalysisNexxim.py:115]:
```suggestion
```

File: pyaedt/application/AnalysisNexxim.py [hunk 5]
```diff
@@ -70,49 +68,51 @@ def __init__(self, application, projectname, designname, solution_type, setup_na
...
+            
         Returns
         -------
-        type
-            list of strin representing Sparameters of excitations
+        list
+            List of strings representing the S parameters of the excitations.
+            
+        Examples
+        --------
```
Comment (PipKat) [inline pyaedt/application/AnalysisNexxim.py:113]:
```suggestion
```

File: pyaedt/application/AnalysisNexxim.py [hunk 6]
```diff
@@ -70,49 +68,51 @@ def __init__(self, application, projectname, designname, solution_type, setup_na
...
+
```
Comment (PipKat) [inline pyaedt/application/AnalysisNexxim.py:114]:
```suggestion
```

File: pyaedt/application/AnalysisNexxim.py [hunk 7]
```diff
@@ -7,32 +7,30 @@
...
-    Class for Circuit Analysis Setup (Nexxim,)
-
-    It is automatically initialized by Application call (like HFSS,
-    Q3D...). Refer to Application function for inputs definition
+    """FieldCircuitAnalysis class.
+    
+    This class is for circuit analysis setup in Nexxim.
+    
+    It is automatically initialized by an application call (like HFSS,
+    Q3D...). See the application function for parameter definitions.
```
Comment (akaszynski) [inline pyaedt/application/AnalysisNexxim.py:15]:
```suggestion
    It is automatically initialized by an application call (e.g. HFSS,
    Q3D, etc.). See the application function for parameter definitions.
```

File: pyaedt/application/AnalysisNexxim.py [hunk 8]
```diff
@@ -128,20 +126,25 @@ def get_all_sparameter_list(self, excitation_names=[]):
...
         -------
-        type
-            list of string representing Return Losses of excitations
+        list
+            List of strings representing the return losses of the excitations.
+            
+        Examples
+        --------
+        
+        >>>excitation_names ["1","2"] output ["S(1,1)",, S(2,2)]
```
Comment (akaszynski) [inline pyaedt/application/AnalysisNexxim.py:147]: Still need to move this example above to the parameters.

File: pyaedt/application/AnalysisNexxim.py [hunk 9]
```diff
@@ -190,21 +193,21 @@ def get_all_insertion_loss_list(self, trlist=[], reclist=[], tx_prefix='', rx_pr
...
+        
         Parameters
         ----------
-        trlist :
-            list of Drivers to include (Default value = [])
-        tx_prefix :
-            prefix for TX (eg. "DIE") (Default value = "")
+        trlist : list, optional
+            List of drivers. The default is ``[]``. For example,
+            ``["1", "2", "3"``.
```
Comment (akaszynski) [inline pyaedt/application/AnalysisNexxim.py:202]:
```suggestion
            ``["1", "2", "3"]``.
```

File: pyaedt/application/AnalysisNexxim.py [hunk 10]
```diff
@@ -190,21 +193,21 @@ def get_all_insertion_loss_list(self, trlist=[], reclist=[], tx_prefix='', rx_pr
...
+        tx_prefix : str, optional
+            Prefix to add to driver names. For example, ``"DIE"``.  The default is ``""``.
 
         Returns
         -------
-        type
-            list of string representing Near End XTalks
+        list
+            List of strings representing near end XTalks of the excitations.
+            For example, ``["S(1,2)", "S(1,3)", "S(2,3)"]``.
```
Comment (akaszynski) [inline pyaedt/application/AnalysisNexxim.py:210]:
```suggestion
            For example, ``["S(1, 2)", "S(1, 3)", "S(2, 3)"]``.
```

File: pyaedt/application/AnalysisNexxim.py [hunk 11]
```diff
@@ -219,29 +222,32 @@ def get_next_xtalk_list(self, trlist=[], tx_prefix=""):
...
-            prefix for RX (eg. "BGA") (Default value = '')
-        skip_same_index_couples :
-            Boolean ignore TX and RX couple with same index (Default value = True)
-
+        trlist : list, optional
+            List of drivers. The default is ``[]``. For example,
+            ``["1", "2"]``.
+        reclist : list, optional
+            List of receiver. The default is ``[]``. For example,
+            ``["3","4"]``.
```
Comment (akaszynski) [inline pyaedt/application/AnalysisNexxim.py:234]:
```suggestion
            ``["3", "4"]``.
```

File: pyaedt/application/AnalysisNexxim.py [hunk 12]
```diff
@@ -219,29 +222,32 @@ def get_next_xtalk_list(self, trlist=[], tx_prefix=""):
...
+            excluded from the list.
+        
         Returns
         -------
-        type
-            list of string representing Far End XTalks
-
+        list
+            List of strings representing the far end XTalks of the excitations.
+            For example, ``["S(1,4)", "S(2,3)"]``.
```
Comment (akaszynski) [inline pyaedt/application/AnalysisNexxim.py:249]:
```suggestion
            For example, ``["S(1, 4)", "S(2, 3)"]``.
```

File: pyaedt/application/AnalysisNexxim.py [hunk 13]
```diff
@@ -70,49 +68,49 @@ def __init__(self, application, projectname, designname, solution_type, setup_na
...
 
         Parameters
         ----------
-        excitation_names :
-            list of excitation to include (Default value = [])
-
+        eexcitation_names : list, optional
+            List of excitations. The default is ``[]``, in which case
+            the S parameters for all excitations are to be provided. 
+            For example, ``["1","2"]``.
```
Comment (akaszynski) [inline pyaedt/application/AnalysisNexxim.py:106]:
```suggestion
            For example, ``["1", "2"]``.
```

File: pyaedt/application/AnalysisNexxim.py [hunk 14]
```diff
@@ -128,20 +126,25 @@ def get_all_sparameter_list(self, excitation_names=[]):
...
         Parameters
         ----------
-        excitation_names :
-            list of excitation to include (Default value = [])
-        excitation_name_prefix :
-             (Default value = '')
-
+        excitation_names : list, optional
+            List of excitations. The default is ``[]``, in which case
+            the return losses for all excitations are to be provided.
```
Comment (akaszynski) [inline pyaedt/application/AnalysisNexxim.py:135]:
```suggestion
            the return losses for all excitations are to be provided.
            For example ``["1", "2"]``.
```

File: pyaedt/application/AnalysisNexxim.py [hunk 15]
```diff
@@ -128,20 +126,25 @@ def get_all_sparameter_list(self, excitation_names=[]):
...
+            the return losses for all excitations are to be provided. 
+        excitation_name_prefix : string, optional
+             Prefix to add to the excitation names. The default is ``""``, 
+             
         Returns
         -------
-        type
-            list of string representing Return Losses of excitations
+        list
+            List of strings representing the return losses of the excitations.
```
Comment (akaszynski) [inline pyaedt/application/AnalysisNexxim.py:142]:
```suggestion
            List of strings representing the return losses of the excitations.
            For example ``["S(1, 1)",, S(2, 2)]``
```

File: pyaedt/application/AnalysisNexxim.py (hunk 8 above)
Comment (akaszynski) [inline pyaedt/application/AnalysisNexxim.py:147]:
```suggestion
```

**Review by akaszynski — APPROVED** [review]
Approved with a few more suggestions.

**Pull request merged**